cmd/sync.go

```diff
@@ -793,4 +829,7 @@ func init() {
 	// Deprecate the old persist-smb-permissions flag
 	syncCmd.PersistentFlags().MarkHidden("preserve-smb-permissions")
 	syncCmd.PersistentFlags().BoolVar(&raw.preservePermissions, PreservePermissionsFlag, false, "False by default. Preserves ACLs between aware resources (Windows and Azure Files, or ADLS Gen 2 to ADLS Gen 2). For Hierarchical Namespace accounts, you will need a container SAS or OAuth token with Modify Ownership and Modify Permissions permissions. For downloads, you will also need the --backup flag to restore permissions where the new Owner will not be the user running AzCopy. This flag applies to both files and folders, unless a file-only filter is specified (e.g. include-pattern).")
+
+	// Limit on size of ObjectIndexerMap in memory.
+	syncCmd.PersistentFlags().StringVar(&raw.maxObjectIndexerMapSizeInGB, "maxObjectIndexerMapSizeInGB", "", "maxObjectIndexerMapSizeInGB is the limit of map size in memory, provide the value in terms of GB")
```

Comment:
"maxObjectIndexerMapSizeInGB" -> "max-indexer-map-gb"

to be inline with other cmdline arguments.